service/api_task.go

```diff
@@ -442,10 +444,15 @@ func assignNextAvailableTask(ctx context.Context, taskQueue *model.TaskQueue, di
 			continue
 		}
 
+		// If the current task group is finished we leave the task on the queue, and indicate the current group needs to be torn down.
```

Comment:
I'm torn about the right thing to do here. Some dispatchers will mark the task dispatched in their in-memory representations.  I think `dispatcher.RefreshFindNextTask` does this. Would this prevent this task from being dispatched? I think @StuartButler might be the best person to ask about this.

Reply:
Stu and @tychoish have both looked at this and it should be fine. I've taken another look since it's been in waiting for so long but nothing new seems to have affected this workflow so I think it's time to merge unless anyone objects.